### PR TITLE
Write indexing arrays into dask graph to reduce size for multiple xarray variables

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/tests/test_order.py [hunk 1]
```diff
@@ -1284,7 +1286,7 @@ def test_anom_mean():
     max_age_transpose = max(ages_tranpose.values())
     assert max_age_transpose < 150
     assert avg_age_transpose < 100
-    assert sum(pressure) / len(pressure) < 100
```
Comment (phofl) [inline dask/tests/test_order.py:1287]: This hovers around 99 on main and now we add one task so we get to approx 100

File: dask/array/_shuffle.py [hunk 2]
```diff
@@ -211,6 +211,8 @@ def _shuffle(chunks, indexer, axis, in_name, out_name, token):
     split_name = f"shuffle-split-{token}"
     slices = [slice(None)] * len(chunks)
     split_name_suffixes = count()
+    sorter_name = "array-sorter-"
+    taker_name = "array-taker-"
```
Comment (hendrikmakait) [inline dask/array/_shuffle.py:215]:
```suggestion
    sorter_name = "shuffle-sorter-"
    taker_name = "shuffle-taker-"
```

**Review by hendrikmakait — APPROVED** [review]
The code looks good to me. I'd like to make sure that this not have a negative performance impact before merging.

**Comment by phofl** [comment]
AB tests look good https://github.com/coiled/benchmarks/actions/runs/10683023390/artifacts/1886066719, I don't trust the speedup though, that seems a bit odd

**Pull request merged**